### PR TITLE
fix: handle missing stack when throwing recursion errors

### PR DESCRIPTION
**Pull request by SimenB**
This is for #398 which I think we should land and release as 8.0.1. Guards against the error and falls back to the behavior of v7

### CODE REVIEW

File: src/fake-timers-src.js [hunk 1]
```diff
@@ -109,6 +109,7 @@ const globalObject = require("@sinonjs/commons").global;
  * @property {number} createdAt
  * @property {boolean} immediate
  * @property {number} id
+ * @property {Error} [error]
```
Comment (SimenB) [inline src/fake-timers-src.js:112]: I _think_ this is how you say `optional`?

Comment (benjamingr) [reply src/fake-timers-src.js:112]: I think so

File: src/fake-timers-src.js [hunk 2]
```diff
@@ -309,14 +310,18 @@ function withGlobal(_global) {
...
+     * @param {Clock} clock
+     * @param {Timer} job
      */
     function getInfiniteLoopError(clock, job) {
         const infiniteLoopError = new Error(
             `Aborting after running ${clock.loopLimit} timers, assuming an infinite loop!`
         );
 
+        if (!job.error) {
+            return infiniteLoopError
```
Comment (benjamingr) [inline src/fake-timers-src.js:322]: semicolon?

**Review by benjamingr — APPROVED** [review]

File: src/fake-timers-src.js (hunk 2 above)
Comment (SimenB) [reply src/fake-timers-src.js:322]: hah, no CI means reviewer needs to notice lint errors 🙈

Comment (SimenB) [reply src/fake-timers-src.js:322]: wait, there's no lint rule for semis 🤦

**Comment by fatso83** [comment]
So ... what's the order of things here? #398 and then #401 and then #402?

**Comment by SimenB** [comment]
I think this, release, then figure out what to do in #398 (I'm not totally happy with it)

**Comment by SimenB** [comment]
I'll rebase both to trigger CI 🙂

**Pull request merged**